src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -972,6 +1008,50 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		return true;
 	}
 
+	/**
+	 * Refresh Azure access token
+	 * @param profile connection profile
+	 */
+	public async refreshAzureAccountTokens(profile: ConnectionProfile): Promise<void> {
+
+		let uri = this.getConnectionUri(profile);
+		//wait for the pending reconnction promise if any
+		const previousReconnectPromise = this._uriToReconnectPromiseMap[uri];
+		if (previousReconnectPromise) {
+			this._logService.info(`Found pending reconnect promise for uri ${uri}, waiting.`);
+			try {
+				const previousConnectionResult = await previousReconnectPromise;
+				if (previousConnectionResult && previousConnectionResult.connected) {
+					this._logService.info(`Previous pending reconnection for uri ${uri} succeeded.`);
+					return;
+				}
+				this._logService.info(`Previous pending reconnection for uri ${uri} failed.`);
+			} catch (err) {
+				this._logService.info(`Previous pending reconnect promise for uri ${uri} is rejected with error ${err}, will attempt to reconnect if necessary.`);
+			}
+		}
+
+
+		this._logService.info(`Timed token refresh for connection ${profile.id} with uri ${uri}`);
```

Comment:
This isn't necessarily fired from the auto-refresh though? I'd just make it more generic like `Refreshing Azure tokens for connection ...`

Reply:
This function only gets called from the timed process

Reply:
That's just by convention though, and you should be working to consolidate this kind of logic anyways (after all it should all be doing the same core logic which is refreshing the token). 

Making the logging messages generic shouldn't be too hard and not really reduce the usefulness of them so doing that would be preferable (along with combining logic when it makes sense)